### DISCUSSION TITLE
How to plot discrete point with matplotlib?

### DISCUSSION THREAD

**Discussion by YanXingYu-star** [hgrecco/pint#1682] [Q&A] [answered]
I am doing some calculation with pint and then I need to plot them with matplotlib.

For a array of data (likes `[1, 1, 1] ureg.liter`), this is fine.

But when I link some independent points (likes `[1 ureg.liter, 1ureg.liter]`), it will return `DimensionalityError: Cannot convert from 'liter' to 'dimensionless'`

there is my code
``` python
import matplotlib.pyplot as plt
import numpy as np
import pint
ureg = UnitRegistry()
ureg.setup_matplotlib()

#some calculation......

va = 0.5 * ureg.L
vc = 0.05 *ureg.L
pa = 0.1013 * ureg.MPa
pb = 2*ureg.MPa
pc = 7*ureg.MPa

a_ = np.arange(180,361,10)
v =np.cos(np.radians(a_))
p = pa*(va)**1.34/(v**1.34)

fig, ax = plt.subplots()
ax.xaxis.set_units(ureg.L)
ax.yaxis.set_units(ureg.MPa)
ax.plot(v, p)  # draw the line -> right

ax.plot([vc,vc],[pb,pc])  #connect two point -> Wrong

```
Did I do some thing wrong? How can I connect the two points?

Thank you!

**Comment by dopplershift** [comment, marked as answer]
When you do `ax.plot([vc,vc],[pb,pc])`, Matplotlib is going to try to convert `[ 0.05 * ureg.L, 0.05 * ureg.L]` and `[2 * ureg.MPa, 7 * ureg.MPa]` to arrays, which is not going to do what you want (at best it will result in an array of objects). You want to create a single array to pass to `plot`, as you did on the line above. One way to do that would be:

```python
ax.plot([vc.m, vc.m] * vc.units, [pb.m, pc.m_as(pb.units)] * pb.units)
```

**Reply by keewis** [reply]
another way would be to use `ureg.force_ndarray_like = True` (not sure if that's actually necessary) and call `np.stack` or `np.concatenate` on the input data:
```python
ax.plot(np.stack([vc, vc]), np.stack([pb, pc]))
```
As far as I remember, that should automatically convert to consistent units.

**Reply by YanXingYu-star** [reply]
Thank you! these are all very helpful.
